pe-parser-library/src/buffer.cpp

```diff
@@ -253,9 +253,30 @@ bounded_buffer *readFileToFileBuffer(const char *filePath) {
   }
 
   p->detail->sec = hMap;
+  
+  LPVOID ptr = nullptr;
+  
+  if (!LargeFile) {
+    ptr = MapViewOfFile(hMap, FILE_MAP_READ, 0, 0, 0);
+  } else {
+    ptr = VirtualAlloc(NULL, fileSize, MEM_COMMIT | MEM_RESERVE, PAGE_READWRITE);
+    if (ptr == INVALID_HANDLE_VALUE) {
+      CloseHandle(h);
+      CloseHandle(ptr);
+      return nullptr;
+    }
+
+    const bool bFileRead = ReadFile(h, ptr, fileSize, nullptr, nullptr);
+    if(!bFileRead) {
+      CloseHandle(h);
+      if (ptr != nullptr) {
+        CloseHandle(ptr);
+      }
+
+      return nullptr;
+    }
```

Comment:
This looks like it only uses a mmap backing on Windows, but we'll probably want similar behavior on other OSes as well. Could you add that to this changeset?